cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,326 @@
+
+```
+rad env update qa --aws-account-id <aws-account-id> --aws-region <aws-region> 
+```
+
+At this point, we can use aws AssumeRole to manage resources using environment specific account information. rp (for recipes) and ucp will have code changes that make sure of aws assumerole to be able to use the right credential specific to the radius environment for managing the aws resources.  
+
+```
+  client := sts.NewFromConfig(cfg)
+  assumeRoleArn := retrieveRoleARNCredential() //has logic to get the roleARN stored as credential
+  // Assume the new role
+  assumeRoleProvider := stscreds.NewAssumeRoleProvider(client, assumeRoleArn)
+  // Create a new config with the new credentials
+  cfg.Credentials = assumeRoleProvider
+  cfg.Region = region
+  // Create a new S3 service client
+  s3client := s3.NewFromConfig(cfg)
+  // Call the ListBuckets function
+  output, err := s3client.ListBuckets(context.TODO(), &s3.ListBucketsInput{})
+  :
+
+```
+
+### API design
+
+```
+@doc("AWS credential kind")
+enum AWSCredentialKind {
+  @doc("The AWS IRSA credential")
+  IRSA,
+}
+```
+
+```
+model AwsAccessKeyCredentialProperties extends AwsCredentialProperties {
+  @doc("IRSA kind")
+  kind: AWSCredentialKind.IRSA;
+ 
+  @doc("The roleARN")
+  roleARN: string;
+
+  @doc("The storage properties")
+  storage: CredentialStorageProperties;
+}
+```
+
+Based on this new model we have to update API calls. 
+
+## User Experience/ CLI Design
+
+`rad init` and `rad credential` commands should be updated to support AWS IRSA.
+Also, `rad install kubernetes --set` should be able to set a AWS role-arn. 
+
+### non interactive support
+rad install kubernetes --set global.aws.IRSA.enabled=true #default false
+
+if enabled, it should print a success/ failure message after updating the pod spec to mount the token. It should also print a information message to the complete of configuration by using rad credential register and rad env update.
+
+rad credential register aws --type <IRSA> --role <roleARN>
+rad credential register aws --type <access-key> --access-key-id <access-key-id> --secret-access-key <secret-access-key>
+
+rad env update --aws-account-id <aws-account-id> --aws-region <aws-region> 
+
+### interactive rad init support
+```
+% rad init --full  
+Enter an environment name 
+>default 
+
+% rad init --full  
+Select your cloud provider                     
+1. Azure                                  
+> 2. AWS                                     
+3.[back] 
+
+% rad init --full  
+
+Select the identity for the AWS cloud provider 
+1. access key                              
+> 2. IRSA  
+
+% rad init --full  
+
+Enter the AWS RoleARN for configuring IRSA: <role-arn-radius>
+
+```
+
+
+Once radius is installed, we use rad env update to update the aws provider config
+```
+% rad env update --aws-account-id <aws-account-id> --aws-region <aws-region> 
```

Comment:
I think rad init should take care of the environment update. we should ask for the account id and region as part of the init flow (I think we do this today)